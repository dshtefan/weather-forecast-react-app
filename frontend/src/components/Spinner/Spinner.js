import React from 'react';
import './Spinner.scss';

const Spinner = () =>
  <div className="lds-css ng-scope">
    <div className="lds-eclipse">
      <div/>
    </div>
  </div>;

export default Spinner;