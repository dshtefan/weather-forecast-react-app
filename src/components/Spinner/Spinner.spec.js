import React from 'react';
import Spinner from './Spinner';

describe('Spinner', () => {
  describe('Компонент содержит необходимые классы', () => {
    const spinner = shallow(<Spinner />);

    it ('Содержит класс: "lds-css"', () => {
      expect(spinner.hasClass('lds-css')).toBeTruthy();
    });

    it ('Содержит класс: "ng-scope"', () => {
      expect(spinner.hasClass('ng-scope')).toBeTruthy();
    });

    it ('Содержит класс: "lds-eclipse"', () => {
      expect(spinner.find('.lds-css').children().hasClass('lds-eclipse')).toBeTruthy();
    });

    it('отображается корректно', () => {
      expect(shallowToJson(spinner)).toMatchSnapshot();
    });
  });

});