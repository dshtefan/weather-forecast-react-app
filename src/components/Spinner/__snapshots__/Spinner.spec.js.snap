// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Компонент содержит необходимые классы отображается корректно 1`] = `
<div
  className="lds-css ng-scope"
>
  <div
    className="lds-eclipse"
  >
    <div />
  </div>
</div>
`;
